src/lib/LedgerApi.ts

```diff
@@ -326,39 +323,57 @@ class LedgerApi {
         }
 
         // prepare tx outside of request to avoid that an error would result in an endless loop in _callLedger
-        const senderPubKeyBytes = await LedgerApi.getPublicKey(keyPath);
+        const [senderPubKeyBytes] = await Promise.all([
+            LedgerApi.getPublicKey(keyPath, walletId),
+            Nimiq.WasmHelper.doImportBrowser() // wasm needed for toAddress
+                .catch((e) => LedgerApi._throwError(LedgerApi.ErrorType.FAILED_LOADING_DEPENDENCIES, e)),
+        ]);
         const senderPubKey = new Nimiq.PublicKey(senderPubKeyBytes);
-        const senderAddress = senderPubKey.toAddress().toUserFriendlyAddress();
-        if (transaction.sender.replace(/ /g, '')
-            !== senderAddress.replace(/ /g, '')) {
+        const senderAddress = senderPubKey.toAddress();
+        if (!senderAddress.equals(transaction.sender)) {
             LedgerApi._throwError(LedgerApi.ErrorType.REQUEST_ASSERTION_FAILED,
-                'Sender Address doesn\'t match this ledger account', request);
+                'Sender Address doesn\'t match this ledger address', request);
         }
+        const senderType = transaction.senderType !== undefined
+            ? transaction.senderType
+            : Nimiq.Account.Type.BASIC;
```

Comment:
These are written on purpose in a more expressive way. Writing `transaction.senderType || Nimiq.Account.Type.BASIC` requires the additional knowledge that not another type than `Nimiq.Account.Type.BASIC` is 0.

Reply:
So if I pass in `null` as `transaction.senderType`, then this is not going to catch it, and furthermore 351 this would lead to the transaction becoming an extended one, although not necessary. Same for `recipientType` and `flags`.

Reply:
Typescript is will complain about the incompatible input value / type `null`.